llama_cpp_guidance/llm.py

```diff
@@ -83,41 +86,41 @@ def session(self, asynchronous=False):
 
     def __call__(self, *args, **kwargs):
         logger.debug("Invoking LlamaCpp ({args}) ({kwargs})", args=args, kwargs=kwargs)
-        output = self.llm(*args, **kwargs)
-        logger.debug(
-            "LlamaCpp generated text: {text} ({choices})",
-            text=output["choices"][0]["text"],
-            choices=[
-                (choice["text"], choice["logprobs"]) for choice in output["choices"]
-            ],
-        )
-
-        for choice in output.get("choices", []):
-            logprobs = choice.get("logprobs")
-
-            if not logprobs:
-                continue
-
-            new_top_logprobs = []
-            for index, top_logprobs in enumerate(logprobs["top_logprobs"]):
-                if top_logprobs is None:
-                    top_logprobs = {choice["logprobs"]["tokens"][index]: -0.1}
-
-                new_top_logprobs.append(top_logprobs)
-            logprobs["top_logprobs"] = new_top_logprobs
-
-        return output
+        for output in self.llm(*args, **kwargs):
+            logger.debug(
+                "LlamaCpp generated text: {text} ({choices})",
+                text=output["choices"][0]["text"],
+                choices=[
+                    (choice["text"], choice["logprobs"]) for choice in output["choices"]
+                ],
+            )
+    
+            for choice in output.get("choices", []):
+                logprobs = choice.get("logprobs")
+    
+                if not logprobs:
+                    continue
+    
+                new_top_logprobs = []
+                for index, top_logprobs in enumerate(logprobs["top_logprobs"]):
+                    if top_logprobs is None:
+                        top_logprobs = {choice["logprobs"]["tokens"][index]: -0.1}
+    
+                    new_top_logprobs.append(top_logprobs)
+                logprobs["top_logprobs"] = new_top_logprobs
+    
+            yield output
```

Comment:
Minor formatting suggestions to prevent trailing whitespace in lines.
```suggestion
        for output in self.llm(*args, **kwargs):
            logger.debug(
                "LlamaCpp generated text: {text} ({choices})",
                text=output["choices"][0]["text"],
                choices=[
                    (choice["text"], choice["logprobs"]) for choice in output["choices"]
                ],
            )

            for choice in output.get("choices", []):
                logprobs = choice.get("logprobs")

                if not logprobs:
                    continue

                new_top_logprobs = []
                for index, top_logprobs in enumerate(logprobs["top_logprobs"]):
                    if top_logprobs is None:
                        top_logprobs = {choice["logprobs"]["tokens"][index]: -0.1}

                    new_top_logprobs.append(top_logprobs)
                logprobs["top_logprobs"] = new_top_logprobs

            yield output
```